healthyregions/SDOHPlace
# #273: Add filter function for spatial resolutions and create the new UI

This PR addresses #273. It includes adding the filter function to the spatial resolution checkboxes and updating the UI to reflect the new design for this section. I also updated a few filter methods to remove redundancy (this will be an ongoing process to optimize the code).

## How to Test
1. Navigate to the search page. No spatial resolution filter should be selected.

2. Select one of the spatial resolution checkboxes. The results list should be updated based on the selection, and the map layer should update accordingly:
<img width="1664" alt="Screenshot 2024-08-28 at 6 08 41 PM" src="https://github.com/user-attachments/assets/e151bac7-f41c-4eed-b00e-2d2506556244">

3. Select multiple spatial resolution checkboxes. The results list should be filtered by any of the selected spatial resolutions, while all relevant layers should be added to the map.
<img width="1457" alt="Screenshot 2024-08-28 at 6 08 56 PM" src="https://github.com/user-attachments/assets/bf07bdb5-cef8-4021-b4e3-3e898a7e0894">

4. Uncheck the checkboxes you selected in steps 2 and 3. The layers and results list should adjust accordingly.

5. Repeat steps 2 to 4 after applying other filters and queries.
<img width="1493" alt="Screenshot 2024-08-28 at 6 09 19 PM" src="https://github.com/user-attachments/assets/7fde99cb-1c4f-4ea3-843a-528924d7ee0a">
